webpack_loader/utils.py

```diff
@@ -17,15 +16,40 @@
     'IGNORE': ['.+\.hot-update.js', '.+\.map']
 }
 
-config.update(getattr(settings, 'WEBPACK_LOADER', {}))
-ignores = [re.compile(I) for I in config['IGNORE']]
+user_config = getattr(settings, 'WEBPACK_LOADER', {})
+
+if not user_config:
+    config = {
+        'DEFAULT': DEFAULT_CONFIG
+    }
+else:
+    config = {}
+    for name, obj in user_config.items():
+        try:
+            entry = dict(DEFAULT_CONFIG, **obj)
+        except TypeError:
+            raise RuntimeError(
+                'Error while parsing WEBPACK_LOADER configuration. '
+                'Is WEBPACK_LOADER config compliant with the new format?')
+
+        entry['ignores'] = [re.compile(I) for I in entry['IGNORE']]
+        config[name] = entry
```

Comment:
Why not something like,

```
user_config = [dict(DEFAULT_CONFIG, **config) for config in user_config.values()]
```

I'm not sure if we should raise a custom error here. It should just fail normally. We should add a django system check to make sure new config is being used. Check https://docs.djangoproject.com/en/1.8/ref/checks/#api-reference

We might have to add apps.py and hook into it's ready method. This will only work on 1.7 and above. It'll fail silently on 1.6. I think that is OK.